controllers/core/core.go

```diff
@@ -34,5 +35,14 @@ func SetupControllers(mgr ctrl.Manager, restClient rest.Interface) (string, erro
 		setupLog.Error(err, "unable to create controller", "controller", "MiniCluster")
 		return "MiniCluster", err
 	}
+	setupLog.Info("🌈 Success controller created", "controller", "MiniCluster")
+
+	if err := (&api.MiniCluster{}).SetupWebhookWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create webhook", "webhook", "MiniCluster")
+		return "MiniCluster", err
```

Comment:
Hmm - so this core.go function is specifically to create controllers. Either we rename it to be about creating resources, or we move the webhook creation to be outside of it, back as its own function from main.go (probably ideal). The error message in main.go that this returns from is referencing a controller failed creation, which could be confusing if it turns out it was a webhook error.

Reply:
Thats good to know, I think the operator sdk originally puts this in main.go, I can't remember why exactly I moved it here

Reply:
Oh I think it would be ok in this file, but maybe in a separate function just to init the webhooks!